sound/soc/sof/topology.c

```diff
@@ -43,6 +43,8 @@
 #define TLV_STEP	1
 #define TLV_MUTE	2
 
+#define SOF_EFFECT_DATA_SIZE 156
```

Comment:
Is this the size of the data required for configuring the effect component?
Why should it be limited to 156? Should it be max ipc size - the headers?

Reply:
it is probably a bad name... It is the position of widget priv + comp data + effect type (where the actual eq payload starts). I understood the comp data needs to be there all the time. Now I'm adding the effect type. As the data following is binary blob, I can't rely on the vendor array mechanism to get to the proper position.

Reply:
I see then maybe it should be called PAYLOAD_OFFSET?

Reply:
also I'm thinking, should this be hardcoded? What if one of the values changes?

Reply:
I mean somehow we need to have an "agreement" what is in the priv member before the payload starts and what is the size of it. I thought before the situation was that for pipeline components you need to have the sof_comp_tokens there. if there's something following that (like bytes field) you somehow need to know the offset to that. I guess the vendor token parsing is updating the  array pointer, but I'm not sure can that be used (is the position guaranteed to be after tuples). I'm open to suggestions :)